### PR TITLE
Introduce `Database` trait

### PR DESCRIPTION
**Pull request by yhabteab**

### CODE REVIEW

File: library/X509/Common/Database.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     * @return  Sql\Connection
+     */
+    protected function getDb()
+    {
+        $config = new Sql\Config(ResourceFactory::getResourceConfig(
+            $this->Config()->get('backend', 'resource')
+        ));
+
+        $options = [
+            \PDO::ATTR_DEFAULT_FETCH_MODE => \PDO::FETCH_OBJ
```
Comment (lippserd) [inline library/X509/Common/Database.php:24]: Please `use PDO;` and what about `PDO::FETCH_BOTH`?

**Review by lippserd — CHANGES_REQUESTED** [review]

**Comment by lippserd** [comment]
And please rebase.

File: library/X509/Common/Database.php (hunk 1 above)
Comment (yhabteab) [reply library/X509/Common/Database.php:24]: > what about `PDO::FETCH_BOTH`?

Wouldn't that be a bit too much? `PDO::FETCH_ASSOC` is only used in the controller, elsewhere we only use `PDO::FETCH_OBJ`.

Comment (lippserd) [reply library/X509/Common/Database.php:24]: > Wouldn't that be a bit too much?

Yes, maybe. Then please add `$options` to getDb() and unify them.

**Review by lippserd — CHANGES_REQUESTED** [review]

File: library/X509/Controller.php [hunk 2]
```diff
@@ -15,29 +16,22 @@
 
 class Controller extends \Icinga\Web\Controller
 {
+    use Database {
+        getDb as private getConn;
```
Comment (lippserd) [inline library/X509/Controller.php:20]: Please call it `getDbWithOptions`.

**Review by lippserd — CHANGES_REQUESTED** [review]

**Review by lippserd — APPROVED** [review]

**Pull request merged**